### DISCUSSION TITLE
BUT... inverter returned no information

### DISCUSSION THREAD

**Discussion by leosmiers** [addiejanssen/domoticz-solaredge-modbustcp-plugin#50] [Q&A]
I am trying to make a connection to my solaredge inverter type: SE4000H-RW000BNN4

I have enabled modbus over tcp in the inverter but I get the following messages in my domoticz log:

```
2022-11-13 11:28:50.766  Status: SolarEdge: Entering work loop.
2022-11-13 11:28:50.766  Status: SolarEdge: Started.
2022-11-13 11:28:51.331  Status: SolarEdge: Initialized version 1.1.1, author 'Addie Janssen'
2022-11-13 11:28:51.335  SolarEdge: Debug logging mask set to: NONE
2022-11-13 11:28:52.408  SolarEdge: Connection established with: solaredge:11502 Device Address: 0. BUT... inverter returned no information
2022-11-13 11:28:52.408  SolarEdge: Retrying to communicate with inverter after: 2022-11-13 11:28:50.330809

```

The last two messages are repeated every 5 seconds.

I use the master branch with solaredge_modbus 0.7.0 and pymodbus 2.5.3

Some extra information. If I run the example program from solaredge_modbus I get the following:

```
pi@domoticz:~/ttt/solaredge_modbus $ python -m example domoticz 11502
Inverter(domoticz:11502, connectionType.TCP: timeout=1, retries=3, unit=0x1):

Registers:
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/pi/ttt/solaredge_modbus/example.py", line 46, in <module>
    print(f"\tManufacturer: {values['c_manufacturer']}")
KeyError: 'c_manufacturer'
```

**Comment by addiejanssen** [comment]
Hi @leosmiers,

Please run the example one more time, but then use the following command line to start it:
```
python -m example --timeout 15 --json domoticz 11502
```

Do you have another application connecting to that inverter already?
These inverters will only allow so many connections at the same time.

Are you sure that the port is `11502` for your inverter?
The default port on my inverter is `502`.
Did you (have to) change the port yourself?

**Comment by leosmiers** [comment]
Hello @Addie
 
Thank you for your reply.
 
I asume you mean: python -m example --timeout 15 --json solaredge11502
(replaced domoticz by solaredge)
 
The command returns an empty json message:
 
{
"meters": {},
"batteries": {
"Battery1": {},
"Battery2": {}
}
}
 
As far as I know there are no other applications connecting to the inverter.

**Reply by addiejanssen** [reply]
So the plugin was correct: there's no information :-)
I do not have an answer to this, but here are some things that people have run into:

Are you sure that the portnumber is correct?
SolarEdge seems to be using different ports for particular inverters.
My inverter is using port 502, but I have also seen reports of port 1502 and 11502.

Is the inverter connected via wifi or cable?
Apparently, there are some inverters that don't support modbus via wifi.

If none of this works to get the example going, you could try to drop the question in the following domoticz forum:
https://www.domoticz.com/forum/viewtopic.php?t=34039&start=60
There are more readers over there then here in my github repo...

**Comment by leosmiers** [comment]
I Hace tried all three mentioned portnumbers (set them also on the inverter).
 
I think I am unlucky in having one that does not work over wifi. I might try it in a later moment when I have more time.
 
Thank you for your support and suggestions.

**Comment by leosmiers** [comment]
Yesterday my inverter has received an update and I am glad to inform that I can now receive the data over a WIFI connection.

thx for the support
